src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/DockerProcessor.kt

```diff
@@ -1,52 +1,56 @@
-    limitations under the License.
-
-    Author: Mikhail Shavkunov (@shavkunov)
+Copyright 2018 BarD Software s.r.o
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Author: Mikhail Shavkunov (@shavkunov)
  */
 package com.bardsoftware.backend.fleet.rmarkdown
 
 import com.spotify.docker.client.DefaultDockerClient
 import com.spotify.docker.client.DockerClient
-import com.spotify.docker.client.DockerClient.LogsParam.stderr
-import com.spotify.docker.client.DockerClient.LogsParam.stdout
 import com.spotify.docker.client.messages.ContainerConfig
-import org.apache.commons.io.FileUtils
+import com.spotify.docker.client.messages.HostConfig
+import org.apache.commons.io.FilenameUtils
 import java.io.File
-import java.nio.charset.Charset
+import java.nio.file.Paths
+
+
+const val PDF_EXTENSION = ".pdf"
 
-class DockerProcessor {
-    private val docker: DockerClient = DefaultDockerClient.fromEnv().build()
+class DockerProcessor(private val docker: DockerClient) {
 
-    fun getMd5Sum(file: File): String {
+    fun compileRmdToPdf(rootFile: File): File {
         var containerId: String? = null
+        val fileName = rootFile.name
+        val parentDir = rootFile.parentFile.absoluteFile.path
 
         try {
-            // TODO: replace this with shell command
-            val content = FileUtils.readFileToString(file, Charset.defaultCharset())
-            val quotedMessage = "\"$content\""
+            val hostConfig = HostConfig.builder()
+                    .appendBinds("$parentDir:/manuscript")
+                    .build()
 
             val containerConfig = ContainerConfig.builder()
-                    .image("busybox")
-                    .cmd("sh", "-c", "echo $quotedMessage | md5sum")
+                    .image("danielak/manuscribble:latest")
+                    .cmd(fileName)
+                    .hostConfig(hostConfig)
                     .build()
 
             val creation = this.docker.createContainer(containerConfig)
             containerId = creation.id()
             this.docker.startContainer(containerId)
+            this.docker.waitContainer(containerId)
 
-            return this.docker.logs(containerId, stdout(), stderr()).use({
-                stream ->stream.readFully() ?: ""
-            })
+            val compiledRmd = FilenameUtils.removeExtension(fileName) + PDF_EXTENSION
+            return Paths.get(parentDir).resolve(compiledRmd).toFile()
         } catch (e: Exception) {
             e.printStackTrace()
```

Comment:
Log it! (or add TODO to log it after merging the neigbor PR)

Reply:
added TODO.